tests/test_blacklist.py

```diff
@@ -40,7 +41,7 @@ def database():
     return db_session
 
 
-TEST_CASES = {
+TEST_CASES: dict[str, Tuple[list[KarmaTransaction], list[KarmaTransaction]]] = {
```

Comment:
Move the type to a variable? IMO, the type is not very helpful here, only exists to placate type checkers. Feels like the sort place to use `auto` in C++.

Reply:
tbh it should have been auto deduced here because it's all literals, not sure why it wasnt. will look again.

Reply:
also just noticed the mix of typing's `Tuple` and base `dict` and `list` :wah: